src/AgDatabase.cs

```diff
@@ -99,8 +99,15 @@ public bool Exists()
     /// </summary>
     public void Delete()
     {
+      _listener.ForEachAgInstance((s, ag) => {
+        if (!ag.IsPrimaryInstance)
+        {
+          ag.Remove(Name);
+          s.Database(Name)?.Drop();
+        }
+      });
       _listener.AvailabilityGroup.Remove(Name);
-      _listener.ForEachAgInstance(s => s.Database(Name)?.Drop());
+      _listener.Primary.Database(Name)?.Drop();
```

Comment:
I think all this can be simplified down to:

```csharp
_listener.ForEachAgInstance((s, ag) => {
    ag.Remove(Name);
    s.Database(Name)?.Drop();
});
```

This looks like it works even if executed on the primary first, tested it locally via this SQL command:
```sql
USE [master]

GO

ALTER AVAILABILITY GROUP [***(redacted)]
REMOVE DATABASE [aniraula_2];

GO

DROP DATABASE [aniraula_2];

Go

```

Reply:
This was my original implementation, but I was seeing the same slow performance while testing. It's been a few days, so I'll doublecheck and report back here what I find.

Reply:
I haven't been able to figure out exactly why, but simplifying the call to the suggestion you made above results in the same exception being thrown. After doing some debugging, I think it may be because the availability database gets dropped on the primary before the call to `LeaveAvailabilityGroup()` on the secondary's.

So to better answer your other question below, I think the problem isn't just making sure we drop the db on the secondary ag before deleting the db, but making sure we drop the db on the secondary ag _before_ we do so on the primary.